### PR TITLE
target masks in rle (json)

### PR DESCRIPTION
**Pull request by taraspiotr**
- change target masks encoding from png to json with rle for masks with borders
- change loss functions so that softmax with cross entropy and sigmoid with bce can use the same targets
- change loaders to handle new targets
- change pipelines to handle different activation functions with different channels configurations

### CODE REVIEW

File: src/loaders.py [hunk 1]
```diff
@@ -53,22 +116,24 @@ def __getitem__(self, index):
...
-        Xi = load_func(self.X, index)
+        Xi = load_func(self.X, index, filetype='png', grayscale=False)
 
         if self.y is not None:
-            Mi = load_func(self.y, index)
+            Mi = self.load_target(self.y, index, load_func)
 
-            Xi, Mi = from_pil(Xi, Mi)
-            Xi, Mi = self.image_augment_with_target(Xi, Mi)
+            Xi, *Mi = from_pil(Xi, *Mi)
```
Comment (jakubczakon) [inline src/loaders.py]: @taraspiotr I think this is cool, it shows the distinction between image and some number of  target masks

File: src/loaders.py [hunk 2]
```diff
@@ -79,19 +144,52 @@ def __getitem__(self, index):
...
 
 
-class ImageSegmentationTTADataset(ImageSegmentationDataset):
+class ImageSegmentationJsonDataset(ImageSegmentationBaseDataset):
+    def load_target(self, data_source, index, load_func):
+        Mi = load_func(data_source, index, filetype='json')
+        return Mi
+
+
+class ImageSegmentationPngDataset(ImageSegmentationBaseDataset):
```
Comment (jakubczakon) [inline src/loaders.py]: @taraspiotr I do feel that this is way cleaner now. How about the performance?
